src/shaders/index.js

```diff
@@ -75,3 +75,63 @@ module.exports = {
         vertexSource: fs.readFileSync(__dirname + '/../shaders/symbol_sdf.vertex.glsl', 'utf8')
     }
 };
+
+// Expand #pragmas to #ifdefs.
+
+const re = /#pragma mapbox: ([\w]+) ([\w]+) ([\w]+) ([\w]+)/g;
+
+for (const programName in module.exports) {
+    const program = module.exports[programName];
+    const fragmentPragmas = {};
+
+    program.fragmentSource = program.fragmentSource.replace(re, (match, operation, precision, type, name) => {
+        fragmentPragmas[name] = true;
+        return operation === 'define' ? `
```

Comment:
I'd favor the more explicit `if (operation === 'define') ... else if (operation === 'initialize')` (or the equivalent `switch`) here and below, so as to (re-)articulate how our custom `#pragma` syntax works.